contracts/src/cosmos/precompile/bank.sol

```diff
@@ -63,4 +65,120 @@ interface IBankModule {
+     */
+    function getSpendableBalanceByDenom(address accountAddress, string calldata denom) external view returns (uint256);
+
+    /**
+     * @dev Returns account balance by address for all denominations.
+     */
+    function getSpendableBalances(address accountAddress) external view returns (Coin[] memory);
+
+    /**
+     * @dev Returns the total supply of a single coin.
+     */
+    function getSupplyOf(string calldata denom) external view returns (uint256);
+
+    /**
+     * @dev Returns the total supply of a all coins.
+     */
+    function getTotalSupply() external view returns (Coin[] memory);
+
+    /**
+     * @dev Returns the parameters of the bank module.
+     */
+    function getParams() external view returns (Param memory);
+
+    /**
+     * @dev Returns the denomination metadata
+     */
+    function getDenomMetadata(string calldata denom) external view returns (DenomMetadata memory);
+
+    /**
+     * @dev Returns all denominations metadata
+     */
+    function getDenomsMetadata() external view returns (DenomsMetadata memory);
+
+    /**
+     * @dev Returns if the denom is enabled to send
+     */
+    function getSendEnabled(string[] calldata denoms) external view returns (SendEnabled memory);
+
+    ////////////////////////////////////// WRITE METHODS //////////////////////////////////////////
+
+    /**
+     * @dev Send coins from one address to another.
+     */
+    function send(address fromAddress, address toAddress, Coin[] calldata amount) external payable returns (bool);
+
+    /**
+     * @dev Send coins from one sender and to a series of different address. 
+     * If any of the receiving addresses do not correspond to an existing account, 
+     * a new account is created.
+     * 
+     * Inputs, despite being `repeated`, only allows one sender input. 
+     */
+    function multiSend(Input calldata input, Output[] calldata outputs) external payable returns (bool);
+
+    //////////////////////////////////////////// UTILS ////////////////////////////////////////////
+
+    /**
+     * @dev Represents a cosmos coin.
+     * Note: this struct is generated as go struct that is then used in the precompile.
+     */
+    struct Coin {
+        uint256 amount;
+        string denom;
+    }
+
+    struct Param {
+        bool defaultSendEnabled;
+    }
+
+    struct DenomUnit {
+        string denom;
+        string[] aliases;
+        uint32 exponent;
+    }
+    struct DenomMetadata {
+        string description;
+        DenomUnit[] denomUnits;
+        string base;
+        string display;
+        string name;
+        string symbol;
+    }
+
+    struct DenomsMetadata {
+        DenomMetadata[] metadatas;
+    }
+
+    struct SendEnabled {
+        string denom;
+        bool enabled;
+    }
+
+    struct Input {
+        address addr;
+        Coin[] coins;
+    }
+    
+    struct Output {
+        address addr;
+        Coin[] coins;
+    }
```

Comment:
rename Input and Output to just 1 struct, like `Balances` or UserBalances or smth